react/GalleryLayout.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { ComponentType, useContext, useMemo } from 'react'
+import classNames from 'classnames'
+
+import { ProductListContext } from 'vtex.product-list-context'
+import { Spinner } from 'vtex.styleguide'
+import { useCssHandles } from 'vtex.css-handles'
+import {
+  useResponsiveValue,
+  MaybeResponsiveInput,
+} from 'vtex.responsive-values'
+import { useRuntime } from 'vtex.render-runtime'
+import { useSearchPageState } from 'vtex.search-page-context/SearchPageContext'
+
+import GalleryLayoutRow from './components/GalleryLayoutRow'
+import SettingsContext from './components/SettingsContext'
+import ProductListEventCaller from './utils/ProductListEventCaller'
+
+export interface LayoutDescription {
+  name: string
+  component: string
+  itemsPerRow: MaybeResponsiveInput<number>
+}
+
+export type Slots = Record<string, ComponentType>
+
+export interface GalleryLayoutProps {
+  layouts: LayoutDescription[]
+  lazyItemsRemaining: number
+  products: any[]
+  showingFacets: boolean
+  summary: any
+  slots: Slots
+}
+
+export type GalleryLayoutPropsWithSlots = Omit<GalleryLayoutProps, 'slots'> &
+  Slots
+
+const LAZY_RENDER_THRESHOLD = 2
+
+const CSS_HANDLES = ['gallery'] as const
+
+const { ProductListProvider } = ProductListContext
+
+const GalleryLayout: React.FC<GalleryLayoutProps> = ({
+  layouts,
+  lazyItemsRemaining,
+  products,
+  showingFacets,
+  summary,
+  slots,
+}) => {
+  const { trackingId = 'Search result' } = useContext(SettingsContext) || {}
+  const handles = useCssHandles(CSS_HANDLES)
+  const { getSettings } = useRuntime()
+  const { galleryLayout } = useSearchPageState()
+
+  const currentLayoutDescription = useMemo(() => {
```

Comment:
I find `description` a bit weird, because it's not really a description is more of a `currentLayoutOption`.

What do you think?

Reply:
I agree, I'll change that :)